src/feedClass.ts

```diff
@@ -0,0 +1,281 @@
+        await gunNewFeed.get('info').get('id').put(id);
+        return gunNewFeed;
+    }
+
+
+    /**
+     * Return user soul
+     */
+    async getOwnerSoul(): Promise<string> {
+        return await this.getSoul(this.gunUser)
+    }
+
+
+    /**
+     * Return soul gun object
+     * @param {Gun} gunInstance
+     */
+    async getSoul(gunInstance: Gun): Promise<string> {
+        return (await gunInstance)['_']['#']
+    }
+
+    /**
+     * Get feed by id
+     *
+     * @param {string} feedInfoName
+     */
+    async getFeed(feedInfoName: string): Promise<Feed | any> {
+        logger.info('class getFeed', {feedInfoName});
+        let result: any = {
+            info: {},
+            items: {},
+            privateItems: {}
+        };
+
+        try {
+            const info = await this.feeds.get(feedInfoName).get('info').then()
+
+            if (info) {
+                result.info = _.pick(info, this.feedInfoFields);
+                result.items = await this.getFeedItems(feedInfoName);
+                result.privateItems = await this.getFeedPrivateItems(feedInfoName)
+            }
+
+        } catch (e) {
+            logger.error(e);
+        }
+
+        return result
+    }
+
+    /**
+     * Return array items
+     *
+     * @param {string} feedInfoName
+     */
+    async getFeedItems(feedInfoName: string): Promise<FeedItem | any> {
+        let result: any = {};
+        logger.info('class getFeedItems', {feedInfoName});
+        const items = await this.feeds.get(feedInfoName).get('items').then();
+        if (items) {
+            for (let keyItem in items) {
+                if (keyItem !== '_') {
+                    const mainFields = await this.feeds
+                        .get(feedInfoName)
+                        .get('items')
+                        .get(keyItem).map();
+                    result[keyItem] = _.pick(mainFields, this.feedItemFields);
+                    const dataItem = await this.feeds
+                        .get(feedInfoName)
+                        .get('items')
+                        .get(keyItem)
+                        .map()
+                        .get('data')
+                    result[keyItem].data = _.pick(dataItem, this.feedItemDataFields)
+                }
+            }
+        }
+        return result
+    }
+
+    /**
+     * Return array items
+     *
+     * @param {string} feedInfoName
+     */
+    async getFeedPrivateItems(feedInfoName: string): Promise<FeedItem | any> {
+        let result: any = {};
+        logger.info('class getFeedPrivateItems', {feedInfoName});
+        const privateItems = await this.feeds.get(feedInfoName).get('privateItems').then();
+        if (privateItems) {
+            for (let privateKey in privateItems) {
+                if (privateKey !== '_') {
+                    result[privateKey] = {};
+                    let items = await this.feeds
+                        .get(feedInfoName)
```

Comment:
there is a .path() shortener allows to avoid log .get() chains
https://gun.eco/docs/API#-a-name-path-a-gun-path-key-